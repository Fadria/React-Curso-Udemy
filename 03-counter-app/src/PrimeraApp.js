import React from 'react';
//import React, { Fragment } from 'react';

const PrimeraApp = () => {
   
    // Forma corta de devolver un Fragment, no sería necesario importar
    return (
        <> 
            <h1>Hola Mundo</h1>
            <p>Mi primera aplicación</p>
        </>
    );

}

export default PrimeraApp;