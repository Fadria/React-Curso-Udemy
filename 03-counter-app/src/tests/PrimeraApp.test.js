import React from 'react';
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe("Pruebas en <PrimeraApp />", () => {
    test('Debe de mostrar el mensaje "Hola desde el index.js"', () => {
        const saludo = "Hola desde el index.js";

        const {getByText} = render(<PrimeraApp saludo={saludo} />);

        expect(getByText(saludo)).toBeInTheDocument();
    })
})