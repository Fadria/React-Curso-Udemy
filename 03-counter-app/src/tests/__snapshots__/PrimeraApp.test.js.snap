// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola desde el index.js
  </h1>
  <h2>
    Buenas tardes
  </h2>
  <h2>
    123.45
  </h2>
  <h2>
    25.22
  </h2>
  <p>
    Mi primera aplicaci√≥n
  </p>
  <p>
    Los booleanos no se imprimen: 
    <Component />
  </p>
  <p>
    1
    2
    3
    4
    5
    6
  </p>
  <pre>
    {
   "nombre": "Federico",
   "edad": 25
}
  </pre>
</Fragment>
`;
